File: relationship-manager/conversation-service/src/main/java/com/bank/rm/conversation/controller/ConversationController.java
package com.bank.rm.conversation.controller;

import com.bank.rm.common.dto.ApiResponse;
import com.bank.rm.conversation.dto.ConversationDtos.*;
import com.bank.rm.conversation.service.ConversationService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.UUID;

@RestController
@RequestMapping("/api/v1/conversations")
public class ConversationController {
    private final ConversationService conversationService;

    public ConversationController(ConversationService conversationService) {
        this.conversationService = conversationService;
    }

    @PostMapping
    public ResponseEntity<ApiResponse<ConversationResponse>> startConversation(
            @RequestBody StartConversationRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED)
                .body(ApiResponse.ok(conversationService.startConversation(request)));
    }

    @PostMapping("/{conversationId}/messages")
    public ResponseEntity<ApiResponse<SendMessageResponse>> sendMessage(
            @PathVariable UUID conversationId, @RequestBody SendMessageRequest request) {
        return ResponseEntity.ok(ApiResponse.ok(conversationService.sendMessage(conversationId, request)));
    }

    @GetMapping("/{conversationId}")
    public ResponseEntity<ApiResponse<ConversationDetailResponse>> getConversation(
            @PathVariable UUID conversationId) {
        return ResponseEntity.ok(ApiResponse.ok(conversationService.getConversation(conversationId)));
    }
}

File: relationship-manager/conversation-service/src/main/kotlin/com/bank/rm/conversation/controller/ConversationController.kt
package com.bank.rm.conversation.controller

import com.bank.rm.common.dto.ApiResponse
import com.bank.rm.conversation.dto.*
import com.bank.rm.conversation.service.ConversationService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.*
import java.util.UUID

@RestController
@RequestMapping("/api/v1/conversations")
class ConversationController(private val conversationService: ConversationService) {

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun startConversation(@Valid @RequestBody request: StartConversationRequest): ApiResponse<ConversationResponse> =
        ApiResponse(success = true, data = conversationService.startConversation(request))

    @PostMapping("/{conversationId}/messages")
    fun sendMessage(
        @PathVariable conversationId: UUID,
        @Valid @RequestBody request: SendMessageRequest,
        authentication: Authentication?
    ): ApiResponse<Map<String, MessageResponse>> {
        val senderId = authentication?.name?.let { UUID.fromString(it) }
        val (customerMsg, aiMsg) = conversationService.sendMessage(conversationId, request, senderId)
        return ApiResponse(
            success = true,
            data = mapOf("customerMessage" to customerMsg, "aiResponse" to aiMsg)
        )
    }

    @GetMapping("/{conversationId}")
    fun getConversationHistory(@PathVariable conversationId: UUID): ApiResponse<ConversationHistoryResponse> =
        ApiResponse(success = true, data = conversationService.getConversationHistory(conversationId))

    @GetMapping("/customer/{customerId}")
    fun getCustomerConversations(@PathVariable customerId: UUID): ApiResponse<List<ConversationResponse>> =
        ApiResponse(success = true, data = conversationService.getCustomerConversations(customerId))

    @PostMapping("/{conversationId}/summarize")
    fun summarize(@PathVariable conversationId: UUID): ApiResponse<ConversationSummaryResponse> =
        ApiResponse(success = true, data = conversationService.summarizeConversation(conversationId))
}
